### PR TITLE
Deprecate PORTABLE_EXE

### LINKED ISSUE #1718: Deprecate PORTABLE_EXE
The keyword `PORTABLE_EXE` is an alias for `EXECUTABLE` when defining forward model jobs for which we have no future plans for 🤷🏻 This points towards that we should probably deprecate and later remove it... Only argument against it would be that it is a breaking change in the configuration. However, it is confusing as is and extremely simple breakage to fix for users.

### PR DESCRIPTION
**Pull request by jondequinor**
Deprecate PORTABLE_EXE

Fixes #1718. Since libres configuration is so broken, it is not
possible to deprecate a _keyword_. Deprecation is on the _item_. So if
PORTABLE_EXE is an alias to EXECUTABLE, deprecating one will deprecate
the other _item_, since they both point to the same one (??).

Even if you could deprecate a keyword, the ext_job configuration
doesn't even print out warnings, since this is done outside of parsing.
This has been fixed---ext_job will now output warnings.

Checks are made that ensures an executable is set, by either keywords.

### CODE REVIEW

**Review by sondreso — APPROVED** [review]
LGTM!

File: libres/lib/job_queue/ext_job.cpp [hunk 1]
```diff
@@ -1020,12 +1019,20 @@ ext_job_type * ext_job_fscanf_alloc(const char * name , const char * license_roo
...
+        } else if (!config_content_has_item(content , EXECUTABLE_KEY) && config_content_has_item(content , "PORTABLE_EXE")) {
+            fprintf(stderr, "%s: PORTABLE_EXE is deprecated. Use %s instead.\n", config_file, EXECUTABLE_KEY);
+            strcpy(exec_key, "PORTABLE_EXE");
         }
 
+        if (ext_job->__valid) {
+            const char * executable     = config_content_get_value_as_executable(content, exec_key);
+            const char * executable_raw = config_content_iget(content, exec_key, 0, 0);
+            ext_job_set_executable(ext_job , executable, executable_raw, search_path);
+        }
```
Comment (sondreso) [inline libres/lib/job_queue/ext_job.cpp:1035]: Should we have a warning if both are set as well (is this even allowed??)? Right now using both will mean that EXECUTABLE_KEY will take priority (which is good), but it also silences the warning about PORTABLE_EXE? :thinking:

Comment (jondequinor) [reply libres/lib/job_queue/ext_job.cpp:1035]: Good catch!

Comment (jondequinor) [reply libres/lib/job_queue/ext_job.cpp]: I've made it so that we at least explain what we've done (ignored PORTABLE_EXE).

Comment (jondequinor) [reply libres/lib/job_queue/ext_job.cpp]: And introduced general warning output in `ext_job`.

**Review by sondreso — APPROVED** [review]
LGTM!

**Pull request merged**